src/Service/NuRaftLogSegment.cpp

```diff
@@ -572,7 +572,8 @@ int NuRaftLogSegment::loadLogEntry(int fd, off_t offset, LogEntryHeader * head,
         return -1;
     }
 
-    entry = LogEntryBody::parse(entry_str, head->term, head->data_length);
+    entry = LogEntryBody::parse(entry_str, head->data_length);
+    entry->set_term(head->term);
 
     delete[] entry_str;
```

Comment:
why not use buffer for entry_str instead of char pointer to avoid copy and delete.

Reply:
Because when we serialize data there is no header of buffer and this copy can not be avoided

Reply:
Got it, maybe better to use `buffer` to avoid manual memory release.

Reply:
No need to worry about memory deallocation issues for the method is only invoked synchronously.

Reply:
But I think we should not use raw pointer in `NuRaftLogSegment.cpp`, maybe fix in another PR.

Reply:
okay